dftimewolf/lib/processors/gcp_cloud_resource_tree_helper.py

```diff
@@ -0,0 +1,303 @@
+
+      return f'projects/{self.project_id}/zones/{self.zone}/{tmp_type}/{self.name}'  # pylint: disable=line-too-long
+
+    return self._resource_name
+
+  @resource_name.setter
+  def resource_name(self, value: str) -> None:
+    """Property resource_name Setter.
+
+       This property setter function will split the resource_name and set the
+       "type", "zone", "project_id" and "name" of the resource.
+
+    Args:
+      value: value to set resource_name to.
+    """
+    # pylint: disable=line-too-long
+    # value example:
+    # 1- If parsing logs: projects/test-project-hkhalifa/zones/us-central1-a/disks/vm1
+    # 2- If querying resource through API call: //www.googleapis.com/compute/beta/projects/test-project-hkhalifa/zones/us-central1-a/disks/vm1
+    # pylint: enable=line-too-long
+    if value:
+      values = value.split('/')
+
+      self.name = values[-1]
+
+      resource_type = values[-2]
+      if resource_type == 'disks':
+        self.type = 'gce_disk'
+      elif resource_type == 'instances':
+        self.type = 'gce_instance'
+      elif resource_type == 'images':
+        self.type = 'gce_image'
+      elif resource_type == 'machineImages':
+        self.type = 'gce_machine_image'
+      elif resource_type == 'instanceTemplates':
+        self.type = 'gce_instance_template'
+      elif resource_type == 'snapshots':
+        self.type = 'gce_snapshot'
+      else:
+        self.type = resource_type
+
+      if '/zones/' in value or '/regions/' in value:
+        self.zone = values[-3]
+        self.project_id = values[-5]
+      elif '/global/' in value:
+        self.zone = 'global'
+        self.project_id = values[-4]
+
+    self._resource_name = value
+
+  @property
+  def creation_timestamp(self) -> Optional[datetime]:
+    """Property creation_timestamp Getter."""
+    return self._creation_timestamp
+
+  @creation_timestamp.setter
+  def creation_timestamp(self, value: Union[datetime, str]) -> None:
+    """Property creation_timestamp Setter.
+
+    Args:
+      value: value to set the creation_timestamp to.
+    """
+    if isinstance(value, datetime):
+      self._creation_timestamp = value
+    else:
+      self._creation_timestamp = parser.parse(value)
+
+  @property
+  def deletion_timestamp(self) -> Optional[datetime]:
+    """Property deletion_timestamp Getter."""
+    return self._deletion_timestamp
+
+  @deletion_timestamp.setter
+  def deletion_timestamp(self, value: Union[datetime, str]) -> None:
+    """Property deletion_timestamp Setter.
+
+    Args:
+      value: value to set the deletion_timestamp to.
+    """
+    if isinstance(value, datetime):
+      self._deletion_timestamp = value
+    else:
+      self._deletion_timestamp = parser.parse(value)
+
+  def IsDeleted(self) -> bool:
+    """Checks if resource is deleted."""
+    if self.deleted or self.deletion_timestamp or not self.creation_timestamp:
+      return True
+
+    return False
```

Comment:
return the condition directly
```suggestion
    return self.deleted or self.deletion_timestamp or not self.creation_timestamp
```

Reply:
Mypy complains when I do that. Also I think it's more readable in the current state.